project/src/main/java/com/google/sps/utilities/SpannerTasks.java

```diff
@@ -701,4 +701,23 @@ public Void run(TransactionContext transaction) throws Exception {
     }
     return results;
   }
+
+  /**
+   * Delete search index entries for given event ID.
+   *
+   * @param eventId eventId to delete search entries for
+   */
+   public static void deleteIndexEntriesByEventId(String eventId) {
+    SpannerClient.getDatabaseClient()
+        .readWriteTransaction()
+        .run(
+            new TransactionCallable<Void>() {
+              @Override
+              public Void run(TransactionContext transaction) throws Exception {
+                String sql = String.format("DELETE FROM Results WHERE EventID=\"%s\";", eventId);
+                long rowCount = transaction.executeUpdate(Statement.of(sql));
+                return null;
```

Comment:
Is `return null` required? (don't actually know, curious)

Reply:
interesting - the return type of public Void run(TransactionContext transaction) throws Exception  is Void. https://www.baeldung.com/java-void-type seems like we can only return null.